core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala

```diff
@@ -401,6 +401,10 @@ class GroupCoordinator(val brokerId: Int,
             val memberMetadata = group.get(memberId)
             responseCallback(memberMetadata.assignment, Errors.NONE)
             completeAndScheduleNextHeartbeatExpiration(group, group.get(memberId))
+
+          case Dead =>
+            // this case is already handled, so raise exception
+            throw new IllegalStateException("can't reach this line")
```

Comment:
Can we provide a more useful message? These impossible states have a funny way of becoming possible over time 😉 . Maybe something like this?
```scala
throw new IllegalStateException(s"Reached unexpected condition for Dead group $groupId")
```
We can probably remove the comment if the exception message is already clear.